src/nvim/eval.c

```diff
@@ -14872,6 +14878,12 @@ static void f_setreg(typval_T *argvars, typval_T *rettv, FunPtr fptr)
                           append, yank_type, block_len);
   }
   rettv->vval.v_number = 0;
+
+  if (set_unnamed) {
+    if (!op_register_set_previous(regname)) {
+      EMSG("Unable to set previous yank register");
```

Comment:
And also note that if you have error you need a test where error is checked unless it is too hard (“too hard” mostly applies to FS errors and races). I have seen lots of bugs in error handling code just because it was not tested; some appeared after the refactoring which did not intend to change functionality.